src/config.rs

```diff
@@ -212,30 +203,19 @@ pub fn get_or_create(config_path: Option<String>) -> Result<Config, String> {
     }
 }
 
-fn path_exists(path: &str) -> bool {
-    std::path::Path::new(path).exists()
-}
-
 pub fn generate_path() -> Result<String, String> {
-    let filename = if cfg!(test) { "test" } else { "tod.cfg" };
-
     let config_directory = dirs::config_dir()
         .ok_or_else(|| String::from("Could not find config directory"))?
         .to_str()
         .ok_or_else(|| String::from("Could not convert config directory to string"))?
         .to_owned();
-    Ok(format!("{config_directory}/{filename}"))
-}
-
-pub fn generate_legacy_path() -> Result<String, String> {
-    let filename = if cfg!(test) { "test" } else { ".tod.cfg" };
-
-    let home_directory = dirs::home_dir()
-        .ok_or_else(|| String::from("Could not find home directory"))?
-        .to_str()
-        .ok_or_else(|| String::from("Could not convert directory to string"))?
-        .to_owned();
-    Ok(format!("{home_directory}/{filename}"))
+    if cfg!(test) {
+        _ = fs::create_dir(format!("{config_directory}/tod_test"));
+        let random_string = Alphanumeric.sample_string(&mut rand::thread_rng(), 30);
+        Ok(format!("tests/{random_string}.testcfg"))
```

Comment:
So now EVERY config created in test gets it's own unique file so testing should be a lot easier. Just create a config and keep referencing that same path for the rest of the test.

It is created in the test directory and gitignored